tmt/base.py

```diff
@@ -375,10 +377,101 @@ def from_spec(cls, raw: _RawRequireFmfId) -> 'RequireFmfId':  # type: ignore[ove
         return fmf_id
 
 
-_RawRequireItem = Union[str, _RawRequireFmfId]
+class _RawRequireFile(TypedDict):
+    type: Optional[str]
+    pattern: Optional[List[str]]
+
+
+@dataclasses.dataclass
+class RequireFile(
+        tmt.utils.SpecBasedContainer,
+        tmt.utils.SerializableContainer,
+        tmt.export.Exportable['RequireFile']):
+    VALID_KEYS: ClassVar[List[str]] = ['type', 'pattern']
+
+    type: Optional[str] = None
+    pattern: Optional[List[str]] = None
```

Comment:
I'd say we could use the default factory here, to always get a list, even an empty one. Can there even be `RequireFile` *without` a `pattern` key?

```python
pattern: List[str] = tmt.utils.field(default_factory=list)
```

Note that the check below, `isinstance(self.pattern, str)`, should never ever succeed: `pattern` is declared as a `Optional[List[str]]`, if it's ever a string, it's a serious error.

I think this is the first such field in requirement classes, we need to use normalization here as well. Note that `FmfId.validate()` does not verify types, it should not be done in `validate()` at all, that's too late. `isinstance(self.pattern, list)` in validate should be *always* true. We want something like this to work transparently like it works in step data and other classes.

```python
pattern: List[str] = tmt.utils.field(
  default_factory=list,
  normalize=tmt.utils.normalize_string_list
)
```